include/lbann/layers/io/input/generic_input_layer.hpp

```diff
@@ -168,15 +169,20 @@ class generic_input_layer : public io_layer {
    * averaged across models. */
   void fp_setup_data(int mini_batch_size) override {
 
-    // Use the predetermined size of the mini-batch to set the current
-    // batch size for the neural network
+    // Determine model mini-batch size and effective mini-batch size
+    // Note: If inter-model communication is activated, the effective
+    // mini-batch is equal to the global mini-batch size.
```

Comment:
Not a criticism of this, more a general comment: Do we need to deal with this elsewhere?

I feel like at least part of this comes down to us abusing the model replication framework for LTFB...